packages/config/utils/utils.js

```diff
@@ -65,6 +87,97 @@ const mapContractToNetworkAddress = network => contracts => {
   }, {})
 }
 
+/**
+ *
+ * @param{string} network
+ * @param{FlowJSONv2.accounts|FlowJSONv2.contracts} objectPossiblyKeyedByNetwork
+ * @returns {Object} { "HelloWorld": "0x123" }
+ */
+const collapseByNetwork = (network, objectPossiblyKeyedByNetwork) => {
+  const result = {}
+  for (const [name, value] of Object.entries(objectPossiblyKeyedByNetwork)) {
+    if (typeof value === "string") {
+      result[name] = value
+      continue
+    }
+
+    if (value[network]) {
+      result[name] = value[network]
+    }
+  }
+
+  return result
+}
+
+/**
+ * @param{string} network
+ * @returns {Object} { "HelloWorld": "0x123" }
+ */
+const resolveV2ContractAliases = network => flowJSON => {
+  const collapsedAccounts = collapseByNetwork(network, flowJSON.accounts)
+  const collapsedContracts = collapseByNetwork(network, flowJSON.contracts)
+
+  const result = {}
+  for (const [contractName, contractValue] of Object.entries(
+    collapsedContracts
+  )) {
+    if (collapsedAccounts[contractValue]) {
+      result[`contracts.${contractName}`] = collapsedAccounts[contractValue]
```

Comment:
Thoughts on pre-pending with something like "system." before the "contracts?" I like the contracts, but one more safety measure in place to avoid conflicts from anything user added could be nice. We could even later down the road make it so anything user added is prepended with "user." Would be cool to have something like WHERE.WHAT.IS if that makes sense?

Reply:
Yeah. That is def doable. Would move the account aliases as well but I'd bet someone in the community is depending on it being there.

Reply:
moved to `system.contracts.*`